controllers/manageddatabase_controller.go

```diff
@@ -436,8 +443,20 @@ func (c *ManagedDatabaseController) reconcileCredentialsForVersion(oneMigration
 		existingDbUsernamesSet.Add(username)
 	}
 
-	// Remove any users that shouldn't be there
+	// Compute the mutations that need to be done to the database
 	dbUsersToRemove := existingDbUsernamesSet.Difference(dbUsernames)
+	dbUsersToAdd := dbUsernames.Difference(existingDbUsernamesSet)
+	secretsToAdd := secretNames.Difference(existingSecretSet)
+
+	numMutationsRequired := dbUsersToAdd.Cardinality() + dbUsersToRemove.Cardinality() + secretsToAdd.Cardinality()
```

Comment:
numMutationsRequired, dbUsersToAdd, secretsToAdd can all be moved inside the conditional below to save work when the operator is not in read only, which should be the common path